packages/dd-trace/test/appsec/rasp/command_injection.integration.spec.js

```diff
@@ -61,28 +62,72 @@ describe('RASP - command_injection - integration', () => {
       }
 
       assert.strictEqual(e.response.status, 403)
-      return await agent.assertMessageReceived(({ headers, payload }) => {
+
+      let appsecTelemetryReceived = false
+
+      const checkMessages = await agent.assertMessageReceived(({ headers, payload }) => {
         assert.property(payload[0][0].meta, '_dd.appsec.json')
-        assert.include(payload[0][0].meta['_dd.appsec.json'], '"rasp-command_injection-rule-id-3"')
+        assert.include(payload[0][0].meta['_dd.appsec.json'], `"rasp-command_injection-rule-id-${ruleId}"`)
       })
+
+      const checkTelemetry = await agent.assertTelemetryReceived(({ headers, payload }) => {
+        const namespace = payload.payload.namespace
+
+        // Only check telemetry received in appsec namespace and ignore others
+        if (namespace === 'appsec') {
+          appsecTelemetryReceived = true
+          const series = payload.payload.series
+          const evalSerie = series.find(s => s.metric === 'rasp.rule.eval')
+          const matchSerie = series.find(s => s.metric === 'rasp.rule.match')
+
+          assert.exists(evalSerie, 'eval serie should exist')
+          assert.include(evalSerie.tags, 'rule_type:command_injection')
+          assert.include(evalSerie.tags, `rule_variant:${variant}`)
+          assert.strictEqual(evalSerie.type, 'count')
+
+          assert.exists(matchSerie, 'match serie should exist')
+          assert.include(matchSerie.tags, 'rule_type:command_injection')
+          assert.include(matchSerie.tags, `rule_variant:${variant}`)
+          assert.strictEqual(matchSerie.type, 'count')
+        }
+      }, 30_000, 'generate-metrics', 2)
+
+      const checks = await Promise.all([checkMessages, checkTelemetry])
+      assert.equal(appsecTelemetryReceived, true)
```

Comment:
in the end you decided not to modify `assertTelemetryReceived`
it is nice this way also.